### ISSUE TITLE
doesn't load fast enough to use as a chat app

### ISSUE THREAD

**Issue by fplord** [igneous-labs/omnispace#12]
every time you enter the page it has to fetch all the rooms which takes a few seconds. this is made worse by the fact that the game client iframe has to reload too, and blocks the main thread.

not sure how exactly to solve this. webapps do lose their context a lot. localStorage caching might help, but not sure how to unblock the main thread

**Comment by billythedummy** [comment]
`matrixClient.startClient()` - 1.78s
![image](https://user-images.githubusercontent.com/12636708/220047730-e628f33e-502c-40fb-8216-07aed49b32e2.png)

It also seems like this function is being called multiple times, with all subsequent requests reaching the max timeout of 30s

**Comment by billythedummy** [comment]
Looked into https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md , seems new/experimental and wasn't just a plug and play - i think the user has to specify the pagination and how to control it and all that. Also, element client doesnt use it so gonna look at other places

**Comment by billythedummy** [comment]
https://github.com/matrix-org/synapse/issues/7781

**Comment by fplord** [comment]
> `matrixClient.startClient()` - 1.78s
> ![image](https://user-images.githubusercontent.com/12636708/220047730-e628f33e-502c-40fb-8216-07aed49b32e2.png)
> 
> It also seems like this function is being called multiple times, with all subsequent requests reaching the max timeout of 30s
> 
> Edit: nvm wrt last line this is because client architecture makes it call the sync endpoint in a loop

1.78s round trip response doesn't seem _too_ bad? as long as first paint is instant (cache chats + make sure loading game client doesn't block)

**Comment by fplord** [comment]
> 1.78s round trip response doesn't seem _too_ bad? as long as first paint is instant (cache chats + make sure loading game client doesn't block)

could consider loading in the iframe with javascript only after the text chats are painted

**Comment by fplord** [comment]
did some experiments today with @billythedummy just now

- (Android) iframe on firefox mobile is blocking, iframe on Chrome mobile is not blocking. With the iframe it takes about 6 seconds to load the messages, without it takes about 2 seconds
- Without the Godot iframe, Hydrogen Web / PWA and our code take aroudn the same amount of time to load
- Is the Godot game client running in debug mode? @f8122dac91

**Comment by f8122dac91** [comment]
@fplord Yeah current CI pipeline setup is exporting debug build to vercel. PR for deploying release build: https://github.com/igneous-labs/telegram-space/pull/47

**Comment by fplord** [comment]
it loads pretty quickly now on raw, quite a nice experience
